CHANGELOG.md

```diff
@@ -3,7 +3,128 @@
+
+The number of EVAL/REDO counts in the profile result are sometimes difficult to understand. This is mainly due to the
+fact that the compiler rewrites expressions and assigns the same location to each generated expression and the profiler
+keys the counters by the location. To provide more clarity, the profile output now includes the number of generated
+expressions for each given expression thereby helping to better understand the result and also how the evaluation works.
+
+Here is an example of the updated profiler output with the new `NUM GEN EXPR` column:
+
+```ruby
++----------+----------+----------+--------------+-------------+
+|   TIME   | NUM EVAL | NUM REDO | NUM GEN EXPR |  LOCATION   |
++----------+----------+----------+--------------+-------------+
+| 20.291µs | 3        | 3        | 3            | test.rego:7 |
+| 1µs      | 1        | 1        | 1            | test.rego:6 |
+| 2.333µs  | 1        | 1        | 1            | test.rego:5 |
+| 6.333µs  | 1        | 1        | 1            | test.rego:4 |
+| 84.75µs  | 1        | 1        | 1            | data        |
++----------+----------+----------+--------------+-------------+
+```
+
+See [the Profiling documentation](https://www.openpolicyagent.org/docs/v0.52.0/policy-performance/#profiling)
+for more details.
+
+Authored by @ashutosh-narkar.
+
+### Runtime, Tooling, SDK
+
+- bundle: Add ability to load bundles from an arbitrary filesystem ([#5833](https://github.com/open-policy-agent/opa/issues/5833)) authored by @kjothen
+- server: Add a note to explicitly point out if OPA binds to the 0.0.0.0 interface on server initialization ([#5090](https://github.com/open-policy-agent/opa/issues/5090)) authored by @Parsifal-M
+- Include trace and span identifier in decision logs to help with correlating logs and trace data ([#5230](https://github.com/open-policy-agent/opa/issues/5230)) authored by @ashutosh-narkar
+
+### Topdown and Rego
+
+- ast: Disallow partial object rules to have other partial object rule within their immediate extent ([#5855](https://github.com/open-policy-agent/opa/issues/5855)) authored by @johanfylling
+- ast: Disallow multi-value rules to have other rules in their extent ([#5813](https://github.com/open-policy-agent/opa/issues/5813)) authored by @johanfylling
+- ast: Set result of groundness check on indexer's AllRules func so that rule evaluation for complete rules is not skipped ([#5857](https://github.com/open-policy-agent/opa/issues/5857)) authored by @ashutosh-narkar
+- rego: Fix duplicate text in error message during module parsing ([#5837](https://github.com/open-policy-agent/opa/pull/5837)) authored by @TzlilSwimmer123
+- planner: Fix bug in call_dynamic with overlapping ref rules ([#5839](https://github.com/open-policy-agent/opa/pull/5839))  authored by @srenatus
+- planner: Fix p.curr <-> prev handling in CallDynamic optimization case ([#5829](https://github.com/open-policy-agent/opa/pull/5829))  authored by @srenatus
```

Comment:
I'd combine these into "planner bugs" that have an impact on IR and Wasm usage. As it stands, it's probably gibberish to most users

Reply:
Can you suggest some text here?

Reply:
Combined the planner bugs. Let me know if the text looks fine.

Reply:
Sorry I hadn't seen your comment earlier. Your text is good, thank you!